baybe/surrogates/base.py

```diff
@@ -56,112 +100,261 @@ class Surrogate(ABC, SerialMixin):
     """Class variable encoding whether or not the surrogate supports transfer
     learning."""
 
-    def to_botorch(self) -> Model:
-        """Create the botorch-ready representation of the model."""
+    _searchspace: SearchSpace | None = field(init=False, default=None, eq=False)
+    """The search space on which the surrogate operates. Available after fitting."""
+
+    # TODO: type should be
+    #   `botorch.models.transforms.outcome.Standardize | _NoTransform`
+    #   but is currently omitted due to:
+    #   https://github.com/python-attrs/cattrs/issues/531
+    _output_scaler = field(init=False, default=None, eq=False)
+    """Scaler for transforming output values. Available after fitting.
+
+    Scales a tensor containing target measurements in computational representation
+    to make them digestible for the model-specific, scale-agnostic posterior logic."""
+
+    def to_botorch(self) -> Model:  # noqa: D102
+        # See base class.
         from baybe.surrogates._adapter import AdapterModel
 
         return AdapterModel(self)
 
-    def posterior(self, candidates: Tensor) -> tuple[Tensor, Tensor]:
-        """Evaluate the surrogate model at the given candidate points.
+    @staticmethod
+    def _make_parameter_scaler_factory(
+        parameter: Parameter,
+    ) -> type[InputTransform] | None:
+        """Return the scaler factory to be used for the given parameter.
 
-        Args:
-            candidates: The candidate points, represented as a tensor of shape
-                ``(*t, q, d)``, where ``t`` denotes the "t-batch" shape, ``q``
-                denotes the "q-batch" shape, and ``d`` is the input dimension. For
-                more details about batch shapes, see: https://botorch.org/docs/batching
+        This method is supposed to be overridden by subclasses to implement their
+        custom parameter scaling logic.
+        """
+        from botorch.models.transforms.input import Normalize
 
-        Returns:
-            The posterior means and posterior covariance matrices of the t-batched
-            candidate points.
+        return Normalize
+
+    @staticmethod
+    def _make_target_scaler_factory() -> type[OutcomeTransform] | None:
+        """Return the scaler factory to be used for target scaling.
+
+        This method is supposed to be overridden by subclasses to implement their
+        custom target scaling logic.
         """
-        import torch
+        from botorch.models.transforms.outcome import Standardize
 
-        # Prepare the input
-        candidates = _prepare_inputs(candidates)
+        return Standardize
 
-        # Evaluate the posterior distribution
-        mean, covar = self._posterior(candidates)
+    def _make_input_scaler(self, searchspace: SearchSpace) -> ColumnTransformer:
```

Comment:
1. Will keep as is because we generally use `make` whenever something is built and `get` when existing stuff is extracted (see also other modules). So changing would mean to break convention.
2. I think this is a bit too nitpicky, tbh, and does not warrant making the name more complicated. You can just as well interpret "make" as "make the scaler that is actually being used", which then includes the fitting as well. But since there was enough space left, I've added `and fit` to the docstring headline.